### PR TITLE
Bug 16308/fix warnings displayed by cargo clippy linter

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/16308

### CODE REVIEW

File: rudder-lang/src/ast/resource.rs [hunk 1]
```diff
@@ -33,7 +33,8 @@ use super::enums::*;
 use super::value::Value;
 use crate::error::*;
 use crate::parser::*;
-use std::collections::{HashMap, HashSet};
+use std::collections::{HashMap, HashSet, hash_map::Entry};
+// use std::collections::hash_map::Entry;
```
Comment (amousset) [inline rudder-lang/src/ast/resource.rs]: no need to keep this line (as it is now part of the other `use` line)

File: rudder-lang/src/ast/enums.rs [hunk 2]
```diff
@@ -556,7 +556,8 @@ impl<'src> EnumList<'src> {
         errors
     }
 
-    /// There can be only one mapping per item that defines an identical descendant
+    // There can be only one mapping per item that defines an identical descendant
```
Comment (amousset) [inline rudder-lang/src/ast/enums.rs]: It there a specific reason for this not being a doc comment? (with `///` instead of `//`)?

Comment (gpoblon) [reply rudder-lang/src/ast/enums.rs]: Not anymore, i'm putting it back. But the following method was unused and before silencing the warning with the `allow` attribute I commented the whole function out... resulting in a compilation error (doc comment right before end of an impl block)

Comment (amousset) [reply rudder-lang/src/ast/enums.rs]: :+1:

File: rudder-lang/src/ast/resource.rs (hunk 1 above)
Comment (gpoblon) [reply rudder-lang/src/ast/resource.rs]: yes, my bad. Wasn't sure about the cleanest way to include it

**Pull request merged**